preprocess.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import pandas as pd
+import processed_sensor_data as psd
+import segment as seg
+import copy
+from scipy import stats, signal, fftpack
+from data_collection import NUM_SENSORS
+
+
+SAMPLING_FREQ = 50
+NYQ_FREQ = SAMPLING_FREQ * 0.5
+
+
+def medfilt(axis):
+    return signal.medfilt(axis)
+
+
+def butter_noise(axis):
+    cutoff_freq = 20.0 / NYQ_FREQ
+    order = 3
+    sos = signal.butter(order, cutoff_freq, btype='lowpass', output='sos')
+    return signal.sosfiltfilt(sos, axis, padlen=0)
+
+
+def butter_grav_body_sep(axis, filter_type):
+    cutoff_freq = 0.3 / NYQ_FREQ
+    order = 3
+    sos = signal.butter(order, cutoff_freq, btype=filter_type, output='sos')
+    return signal.sosfiltfilt(sos, axis, padlen=0)
+
+
+def butter_gravity(axis):
+    return butter_grav_body_sep(axis, filter_type='lowpass')
+
+
+def butter_body(axis):
+    return butter_grav_body_sep(axis, filter_type='highpass')
+
+
+def filter_data(filter, data):
+    return np.apply_along_axis(filter, 0, data)
+
+
+def preprocess_segment(segment, noise_filters):
+    """Takes in a segment and returns it with processed sensor data.
+
+    Each segment is first noise-filtered then acc data is split into
+    bodyAcc and gravAcc data.
+
+    The noise filters are applied in left to right order.
+    """
+    processed_sensors_data = []
+
+    for i in range(NUM_SENSORS):
+        acc = segment.sensors_data[i].acc
+        gyro = segment.sensors_data[i].acc
+
+        for noise_filter in noise_filters:
```

Comment:
I take it that you're not too fond of the `compose` pattern?

IMHO, `compose` makes the code both more readable and more concise and in addition also helps when writing test code (see below)